### PR TITLE
HTML publications republisher

### PR DESCRIPTION
**Pull request by boffbowsh**
Converts the HTML attachment pushing class to a worker suitable for use by both the Edition Services workflow and the DocumentRepublisher worker.

When republishing the parent publications and consultations, the attached HTML publications will be sent to the publishing api in the correct way.

Best reviewed commit by commit.

### CODE REVIEW

File: app/workers/publishing_api_html_attachments_worker.rb [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+  def content_ids_to_remove
+    return Set[] unless previous_edition
+    old_content_ids = previous_edition.html_attachments.pluck(:content_id).to_set
+    new_content_ids = current_html_attachments.pluck(:content_id).to_set
+
+    old_content_ids - new_content_ids
+  end
+
+  def do_publish(update_type)
```
Comment (gpeng) [inline app/workers/publishing_api_html_attachments_worker.rb]: I think the ordering here will cause the issue we've seen where users are trying to create an attachment with the same base path as a previously deleted one and it gets redirected. Should we redirect the `content_ids_to_remove` first and then the new attachment route will overwrite the redirect if one gets created?

File: test/unit/workers/publishing_api_document_republishing_worker_test.rb [hunk 2]
```diff
@@ -19,6 +19,10 @@ class PublishingApiDocumentRepublishingWorkerTest < ActiveSupport::TestCase
     PublishingApiDraftWorker.expects(:new).returns(draft_worker = mock)
     draft_worker.expects(:perform).with(draft_edition.class.name, draft_edition.id, "republish", "en")
 
+    invocation_order = sequence('invocation_order')
+    PublishingApiHtmlAttachmentsWorker.any_instance.expects(:perform).with(published_edition.id, "republish").in_sequence(invocation_order)
```
Comment (gpeng) [inline test/unit/workers/publishing_api_document_republishing_worker_test.rb]: Can we split these lines?

File: test/unit/workers/publishing_api_document_republishing_worker_test.rb [hunk 3]
```diff
@@ -93,6 +97,10 @@ class DraftException < StandardError; end
     PublishingApiUnpublishingWorker.expects(:new).returns(unpublishing_worker = mock)
     unpublishing_worker.expects(:perform).with(published_edition.unpublishing.id, false)
 
+    invocation_order = sequence('invocation_order')
+    PublishingApiHtmlAttachmentsWorker.any_instance.expects(:perform).with(published_edition.id, "republish").in_sequence(invocation_order)
```
Comment (gpeng) [inline test/unit/workers/publishing_api_document_republishing_worker_test.rb]: ...and these.

File: app/workers/publishing_api_html_attachments_worker.rb (hunk 1 above)
Comment (boffbowsh) [inline app/workers/publishing_api_html_attachments_worker.rb]: Good catch. Fixed.

File: test/unit/workers/publishing_api_document_republishing_worker_test.rb (hunk 2 above)
Comment (boffbowsh) [inline test/unit/workers/publishing_api_document_republishing_worker_test.rb]: As you wish

File: test/unit/workers/publishing_api_document_republishing_worker_test.rb (hunk 3 above)
Comment (boffbowsh) [inline test/unit/workers/publishing_api_document_republishing_worker_test.rb]: Done

**Pull request merged**